test.disabled/ejabberd_tests/tests/push_SUITE.erl

```diff
@@ -0,0 +1,715 @@
+            escalus:send(Alice, escalus_stanza:presence(<<"unavailable">>)),
+
+            %% Disable
+            escalus:send(Alice, disable_stanza(<<"pubsub@localhost">>, <<"NodeId">>)),
+            escalus:assert(is_result, escalus:wait_for_stanza(Alice)),
+
+            Msg = <<"Heyah!">>,
+            Stanza = escalus_stanza:groupchat_to(room_bin_jid(Room), Msg),
+            escalus:send(Bob, Stanza),
+
+            ?assert(not truly(received_route())),
+            ok
+        end).
+
+
+%%--------------------------------------------------------------------
+%% Test helpers
+%%--------------------------------------------------------------------
+
+%% ----------------------------------
+%% Stanzas
+%% ----------------------------------
+
+disable_stanza(JID, undefined) ->
+    disable_stanza([
+        {<<"xmlns">>, <<"urn:xmpp:push:0">>},
+        {<<"jid">>, JID}
+    ]);
+disable_stanza(JID, Node) ->
+    disable_stanza([
+        {<<"xmlns">>, <<"urn:xmpp:push:0">>},
+        {<<"jid">>, JID},
+        {<<"node">>, Node}
+    ]).
+disable_stanza(JID) when is_binary(JID) ->
+    disable_stanza(JID, undefined);
+disable_stanza(Attrs) when is_list(Attrs) ->
+    escalus_stanza:iq(<<"set">>, [#xmlel{name = <<"disable">>, attrs = Attrs}]).
+
+enable_stanza(JID, Node) ->
+    enable_stanza(JID, Node, undefined).
+enable_stanza(JID, Node, FormFields) ->
+    enable_stanza(JID, Node, FormFields, ?NS_PUBSUB_PUB_OPTIONS).
+enable_stanza(JID, Node, FormFields, FormType) ->
+    escalus_stanza:iq(<<"set">>, [#xmlel{name = <<"enable">>, attrs = [
+        {<<"xmlns">>, <<"urn:xmpp:push:0">>},
+        {<<"jid">>, JID},
+        {<<"node">>, Node}
+    ], children = maybe_form(FormFields, FormType)}]).
+
+maybe_form(undefined, _FormType) ->
+    [];
+maybe_form(FormFields, FormType) ->
+    [make_form([{<<"FORM_TYPE">>, FormType} | FormFields])].
+
+make_form(Fields) ->
+    #xmlel{name = <<"x">>, attrs = [{<<"xmlns">>, ?NS_XDATA}, {<<"type">>, <<"submit">>}],
+           children = [make_form_field(Name, Value) || {Name, Value} <- Fields]}.
+
+make_form_field(Name, Value) ->
+    #xmlel{name = <<"field">>,
+           attrs = [{<<"var">>, Name}],
+           children = [#xmlel{name = <<"value">>, children = [#xmlcdata{content = Value}]}]}.
+
+
+
+%% ----------------------------------
+%% Other helpers
+%% ----------------------------------
+
+parse_form(#xmlel{name = <<"x">>} = Form) ->
+    parse_form(exml_query:subelements(Form, <<"field">>));
+parse_form(Fields) when is_list(Fields) ->
+    lists:map(
+        fun(Field) ->
+            {exml_query:attr(Field, <<"var">>),
+             exml_query:path(Field, [{element, <<"value">>, cdata}])}
+        end, Fields).
+
+%% @doc Forwards all erlcloud_sns:publish calls to local PID as messages
+start_publish_listener(Config) ->
+    TestCasePid = self(),
+    rpc(meck, new, [ejabberd_router, [no_link, passthrough]]),
+    rpc(meck, expect,
```

Comment:
I don't like the tampering with `ejabberd_router` here. Have you considered declaring e.g. `push.localhost` as JID in `enable` request and registering CT process as domain handler with `register_route`?